src/core/model/reducer.ts

```diff
@@ -72,10 +72,19 @@ export const uiSchemaReducer = (
         newUiSchema.elements.splice(action.index, 0, newUIElement);
         return getRoot(newUiSchema as EditorUISchemaElement);
       });
-    case SET_UISCHEMA_OPTIONS:
-      return withCloneTree(action.uiSchema, uiSchema, (newUiSchema) => {
-        newUiSchema.options = action.options;
-        return getRoot(newUiSchema as EditorUISchemaElement);
+    case UPDATE_UISCHEMA_ELEMENT:
+      return withCloneTree(action.uiSchemaElement, uiSchema, (newUiSchema) => {
+        const updatedUISchemaElement = {
+          ...newUiSchema,
+          ...action.changedProperties,
+        };
+        const parent = newUiSchema.parent as EditorLayout;
+        if (parent) {
+          const index = parent.elements.indexOf(newUiSchema);
+          // update element in its parent element list
+          parent.elements[index] = updatedUISchemaElement;
+        }
```

Comment:
Your parent could also be a control in the case when we drop a Group as the root element of an array control.